### DISCUSSION TITLE
Please describe all ejbca-ce container environment variables

### DISCUSSION THREAD

**Discussion by eryabies** [Keyfactor/ejbca-ce#197] [Ideas]
Hi!

Can you please add the full list of environment variables and their meaning which are used and can be overridden when using the official ejbca-ce container (https://hub.docker.com/r/keyfactor/ejbca-ce)

Thanks!

**Comment by primetomas** [comment]
It already says there under "Other Environment Variables" how you can see the variables. If you have specific questions, don't hesitate to ask.

**Reply by eryabies** [reply]
But inside ejbca-ce container under /opt/primekey/bin/internal there are many different variables and not to much information about what they exactly doing.
For example:
`environment-app:if [ "${TLS_SETUP_ENABLED}" == "later"`

My point is that maybe its good idea to map all these variables outside the shell scripts;)

**Comment by primetomas** [comment]
That one is either the "simple" as defined in the quick start ephemeral instance, or the default as you looked up, and is displayed with 'docker inspect keyfactor/ejbca-ce:latest'. (like TLS_SETUP_ENABLED=true).
Most are described in the docs. Some are self explanatory, and some are internal dev ones, and perhaps some can be better documented. 

From the ones you find in 'docker inspect' is there any that you think need to be better documented?

Like the SMTP ones, SMTP_USERNAME and PASSWORD are not described, but should be almost self explanatory, would you agree?

I understand if TLS_SETUP_ENABLED is not obvious, it could be better described, I agree.

**Reply by mickgarland** [reply]
I agree the SMTP ones are self-explanatory, but they say isn't it all about documentation documentation documentation??

**Reply by primetomas** [reply]
You are right. I'll work on trying to add more to the docs, starting next week when work is getting back to normal.

**Comment by primetomas** [comment]
I updated the documentation with many more variables now. Check out https://hub.docker.com/r/keyfactor/ejbca-ce and let me know what's missing.
